components/CreateOrUpdateRecipeForm.js

```diff
@@ -0,0 +1,232 @@
+        />
+
+        <label htmlFor="imgurl">Image Url</label>
+        <input
+          type="url"
+          id="imgurl"
+          name="imgurl"
+          defaultValue={recipeDetails.imageURL}
+        />
+
+        <StyledFieldSet>
+          <legend>Ingredients:</legend>
+          <input
+            type="text"
+            name="amount1"
+            placeholder="Amount"
+            required
+            defaultValue={
+              recipeDetails.ingredients && recipeDetails.ingredients.length > 0
+                ? recipeDetails.ingredients[0].quantity
+                : ""
+            }
+          />
+          <input
+            type="text"
+            name="ingredient1"
+            placeholder="Ingredient"
+            required
+            defaultValue={
+              recipeDetails.ingredients && recipeDetails.ingredients.length > 0
+                ? recipeDetails.ingredients[0].name
+                : ""
+            }
+          />
+          <input
+            type="text"
+            name="amount2"
+            placeholder="Amount"
+            defaultValue={
+              recipeDetails.ingredients && recipeDetails.ingredients.length > 0
+                ? recipeDetails.ingredients[1].quantity
+                : ""
+            }
+          />
+          <input
+            type="text"
+            name="ingredient2"
+            placeholder="Ingredient"
+            defaultValue={
+              recipeDetails.ingredients && recipeDetails.ingredients.length > 0
+                ? recipeDetails.ingredients[1].name
+                : ""
+            }
+          />
+
+          <input
+            type="text"
+            name="amount3"
+            placeholder="Amount"
+            defaultValue={
+              recipeDetails.ingredients && recipeDetails.ingredients.length > 0
+                ? recipeDetails.ingredients[2].quantity
+                : ""
+            }
+          />
+          <input
+            type="text"
+            name="ingredient3"
+            placeholder="Ingredient"
+            defaultValue={
+              recipeDetails.ingredients && recipeDetails.ingredients.length > 0
+                ? recipeDetails.ingredients[2].name
+                : ""
+            }
+          />
+
+          <input
+            type="text"
+            name="amount4"
+            placeholder="Amount"
+            defaultValue={
+              recipeDetails.ingredients && recipeDetails.ingredients.length > 0
+                ? recipeDetails.ingredients[3].quantity
+                : ""
+            }
+          />
+          <input
+            type="text"
+            name="ingredient4"
+            placeholder="Ingredient"
+            defaultValue={
+              recipeDetails.ingredients && recipeDetails.ingredients.length > 0
+                ? recipeDetails.ingredients[3].name
+                : ""
+            }
+          />
+        </StyledFieldSet>
```

Comment:
This code feels a little repetitive. Are there always 4 ingredients? Why not map over the ingredients array and create an element for each item in the array.

Reply:
@onemanwenttomow Thank you for your review! This made a lot cleaner.

Reply:
@onemanwenttomow Thank you for your review! This made a lot of stuff much cleaner. 

Yes the ingredients are repetitive. We will rework them at a extra [user story](https://github.com/feivlinus/fork-yeah/issues/16). The form would have been different at the beginning and we were suggested to sort the "one or more ingredients part" to a extra user story by Jessica. So we made it 4 by default and the first has to filled out till we start the extra user story for it.